File: src/main/javabak/machine_abstractions/EnergyCapableBlockEntity.java
package redcrafter07.processed.block.machine_abstractions;

import net.minecraft.world.level.block.state.BlockState;
import net.neoforged.neoforge.energy.IEnergyStorage;

import javax.annotation.Nullable;

public interface EnergyCapableBlockEntity {
    @Nullable
    IEnergyStorage energyCapabilityForSide(@Nullable BlockSide side, BlockState state);
}

File: src/main/kotlin/redcrafter07/processed/block/machine_abstractions/EnergyCapableBlockEntity.kt
package redcrafter07.processed.block.machine_abstractions

import net.minecraft.world.level.block.state.BlockState
import redcrafter07.processed.ProcessedPower

interface EnergyCapableBlockEntity {
    fun energyCapabilityForSide(side: BlockSide?, state: BlockState): ProcessedPower?
}